synapse/rest/client/v1/login.py

```diff
@@ -319,9 +319,9 @@ def __init__(self, hs: "HomeServer"):
         # register themselves with the main SSOHandler.
         if hs.config.cas_enabled:
             hs.get_cas_handler()
-        elif hs.config.saml2_enabled:
+        if hs.config.saml2_enabled:
             hs.get_saml_handler()
-        elif hs.config.oidc_enabled:
+        if hs.config.oidc_enabled:
```

Comment:
Is this not a fixing something real?

Reply:
ah, not really. In a real synapse, the OIDC and SAML handlers get instantiated via other codepaths, so it only bit in the test environment.

Reply:
Ah, fair

Reply:
(this code is just belt-and-braces to make sure that the handlers definitely get instantiated if we are on a worker that hosts this endpoint.)